src/ralph/models/selector.py

```diff
@@ -152,3 +161,55 @@ def get_decision_tree(self, model_rules):
         false_subtree = self.get_decision_tree(false_subtree) if false_subtree else None
 
         return {root_rule: {True: true_subtree, False: false_subtree}}
+
+    def validate(self, input_file: TextIO, ignore_errors: bool, fail_on_unknown: bool):
+        """Validates JSON event strings line by line."""
+
+        count = 0
+        success_count = 0
+        for event_str in input_file:
+            try:
+                count += 1
+                yield self._validate_event(event_str)
+                success_count += 1
+            except (json.JSONDecodeError, TypeError) as err:
+                message = "Input event is not a valid JSON string!"
+                self._log_error(message, event_str, err)
+                if not ignore_errors:
+                    raise BackendException(message) from err
+            except UnknownEventException as err:
+                if fail_on_unknown:
+                    raise err
+            except ValidationError as err:
+                message = f"Input event is not a valid {err.model.__name__}!"
+                self._log_error(message, event_str, err)
+                if not ignore_errors:
+                    raise BackendException(message) from err
+        logger.info(
+            "Total events: %d, Invalid events: %d", count, count - success_count
+        )
+
+    def _validate_event(self, event_str: str):
+        """Validates a single JSON string event.
+
+        Raises:
+            TypeError: when the event_str is not of type string.
+            JSONDecodeError: when the event_str is not a valid JSON string.
+            UnknownEventException: when no matching model is found for the event.
+            ValidationError: when the the event is failing the pydantic model validation.
+        Returns:
+            event_str (str): The unchanged input event_str.
+        """
+        event = json.loads(event_str)
+        model = self.get_model(event)
+        if not model:
+            message = "No matching pydantic model found for input event."
+            self._log_error(message, event_str, "model is None")
+            raise UnknownEventException(message)
+        model(**event)
+        return event_str
```

Comment:
Why not returning serialized event?

```suggestion
        return model(**event)
```

Reply:
Good question) I think there are benefits and tradeoffs on both sides:

**If we return the parsed event:**

`pro` pydantic will try to adapt the event so it matches the model (soft validation) and fix minor mistakes. (ex. transform str to int)
`contra` serializing pydantic models cost some additional processing
`contra` we will have to change "ServerEventModel" to not use the "Json" type as it transforms the value back to a dictionary (not the original JSON string) and validating twice the same valid event (in a pipe one after another) will work the first time but will fail the second time.

**If we return the event as they are:**
 
`pro` no additional processing
`pro` we don't change the event (only indicate if it's valid or not)
`contra` we keep the minor mistakes and don't benefit of pydantic soft validation fixes.

For me both should work, which way do you prefer ?

Reply:
Fixing typing errors seems essential for us as for example a database backend such as Elasticsearch will have to infer types on-the-fly to the input events. Having floats or integers instead of strings seems crucial to me. WDYT?

Reply:
Sounds good for me) will change this) Thanks)

Reply:
In the second contra of returning the parsed event, I don't get why the second validation of a same valid event will fail the second time

Reply:
For the server event (`event` field) we use the `Json` pydantic validator (I'm changing that now).
The first time Pydantic validates a server event, it takes the JSON encoded `event` field string `"{\"GET\": {}, \"POST\": {}}"` and creates a dictionary: `{"GET": {}, "POST": {}}`, when it serializes back the event to JSON it keeps the dictionary structure.
The second time Pydantic validates the previous server event it fails because it expects a JSON encoded string but now has a dictionary for the `event` field.

Reply:
Ok I got it! Thanks

Reply:
I think returning the parsed event is better for several points:
* as we tend to do, it's more pythonic for our code*
* the type and format of fields in edx events are very specific for each one of them, I would consider this very benificial to process them with Python object 
* in ralph process, returning `event_str` after running a validation would be generally time consuming for us at the end. We loose information by not keeping Pydantic fixes